src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppEcsFargate/Generated/RecipeConstruct.cs

```diff
@@ -0,0 +1,342 @@
+                throw new ArgumentNullException(nameof(AppVpc));
+
+            var settings = recipeConfiguration.Settings;
+            if (settings.ECSCluster.CreateNew)
+            {
+                EcsCluster = new Cluster(this, nameof(EcsCluster), InvokeCustomizeCDKPropsEvent(nameof(EcsCluster), this, new ClusterProps
+                {
+                    Vpc = AppVpc,
+                    ClusterName = settings.ECSCluster.NewClusterName
+                }));
+            }
+            else
+            {
+                EcsCluster = Cluster.FromClusterAttributes(this, nameof(EcsCluster), InvokeCustomizeCDKPropsEvent(nameof(EcsCluster), this, new ClusterAttributes
+                {
+                    ClusterArn = settings.ECSCluster.ClusterArn,
+                    ClusterName = ECSFargateUtilities.GetClusterNameFromArn(settings.ECSCluster.ClusterArn),
+                    SecurityGroups = new ISecurityGroup[0],
+                    Vpc = AppVpc
+                }));
+            }
+
+            AppTaskDefinition = new FargateTaskDefinition(this, nameof(AppTaskDefinition), InvokeCustomizeCDKPropsEvent(nameof(AppTaskDefinition), this, new FargateTaskDefinitionProps
+            {
+                TaskRole = AppIAMTaskRole,
+                Cpu = settings.TaskCpu,
+                MemoryLimitMiB = settings.TaskMemory
+            }));
+
+            AppLogging = new AwsLogDriver(InvokeCustomizeCDKPropsEvent(nameof(AppLogging), this, new AwsLogDriverProps
+            {
+                StreamPrefix = recipeConfiguration.StackName
+            }));
+
+            if (string.IsNullOrEmpty(recipeConfiguration.ECRRepositoryName))
+                throw new InvalidOrMissingConfigurationException("The provided ECR Repository Name is null or empty.");
+
+            EcrRepository = Repository.FromRepositoryName(this, nameof(EcrRepository), recipeConfiguration.ECRRepositoryName);
+            AppContainerDefinition = AppTaskDefinition.AddContainer(nameof(AppContainerDefinition), InvokeCustomizeCDKPropsEvent(nameof(AppContainerDefinition), this, new ContainerDefinitionOptions
+            {
+                Image = ContainerImage.FromEcrRepository(EcrRepository, recipeConfiguration.ECRImageTag),
+                Logging = AppLogging
+            }));
+
+            AppContainerDefinition.AddPortMappings(new PortMapping
+            {
+                ContainerPort = 80,
+                Protocol = Amazon.CDK.AWS.ECS.Protocol.TCP
+            });
+
+            WebAccessSecurityGroup = new SecurityGroup(this, nameof(WebAccessSecurityGroup), InvokeCustomizeCDKPropsEvent(nameof(WebAccessSecurityGroup), this, new SecurityGroupProps
+            {
+                Vpc = AppVpc,
+                SecurityGroupName = $"{recipeConfiguration.StackName}-ECSService"
+            }));
+
+            var ecsServiceSecurityGroups = new List<ISecurityGroup>();
+            ecsServiceSecurityGroups.Add(WebAccessSecurityGroup);
```

Comment:
is it suppose to reference the instance version of `EcsServiceSecurityGroups`?

```suggestion
            EcsServiceSecurityGroups.Add(WebAccessSecurityGroup);
```

Reply:
Fixed